.travis.yml

```diff
@@ -15,7 +15,22 @@ addons:
     chrome: stable
 install:
     - npm install
-# install synapse prerequisites for end to end tests
-    - sudo apt-get install build-essential python2.7-dev libffi-dev python-pip python-setuptools sqlite3 libssl-dev python-virtualenv libjpeg-dev libxslt1-dev
-script:
-    ./scripts/travis.sh
+    - ./scripts/travis/install-deps.sh
+
+matrix:
+    include:
+        - name: Linting Checks
+          script:
+            # run the linter, but exclude any files known to have errors or warnings.
+            - npm run lintwithexclusions
+        - name: End-to-End Tests
+          install:
+            - npm install
```

Comment:
`install-deps.sh` already contains `npm install`

Reply:
Good point! However, the nice thing about pulling this out of the script is that travis will time how long the `npm install` took *and* how long `./scripts/travis/install-deps.sh` took. The `npm install` in `install-deps.sh` goes a lot faster since its only installing the new dependencies after we've fiddled.

Note that `npm install` in `install-deps.sh` is run against the `matrix-js-sdk` repo.

Reply:
![screenshot from 2019-01-09 11-47-56](https://user-images.githubusercontent.com/8428120/50897575-74972d00-1404-11e9-83cf-448646fa2c46.png)

Is what it looks like in the travis logs